presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/thrift/InMemoryHiveMetastore.java

```diff
@@ -227,9 +227,18 @@ public synchronized void dropTable(MetastoreContext metastoreContext, String dat
         if (deleteData && table.getTableType().equals(MANAGED_TABLE.name())) {
             for (String location : locations) {
                 if (location != null) {
-                    File directory = new File(new Path(location).toUri());
-                    checkArgument(isAncestor(directory, baseDirectory), format("Table directory %s must be inside of the metastore base directory %s", directory, baseDirectory));
-                    deleteDirectory(directory);
+                    try {
+                        File directory = new File(new Path(location).toUri());
+                        checkArgument(isAncestor(directory, baseDirectory), format("Table directory %s must be inside of the metastore base directory %s", directory, baseDirectory));
+                        deleteDirectory(directory);
+                    }
+                    catch (IllegalArgumentException ex) {
+                        // expand exception message to include the undeletable location
```

Comment:
So what was confusing to me about the code is that it looks like we're catching the illegal argument exception we just threw from checkArgument (which we are, but it's because there are other things here that can also throw illegal argument exception that we want to do something with). Maybe it should be obvious, but I was hoping for a comment explaining that the reason for catching IllegalArgumentException is not from the checkArgument, but to augment the error messages coming from IllegalArgumentExceptions from other calls.

Reply:
Oh bleah. I see what you're saying now. The IAE I want to rewrite is from line 231 and I shouldn't be catching the one from line 232. I need to split this out further.